src/AutoLaunchLinux.coffee

```diff
@@ -10,30 +10,39 @@ module.exports =
         file = @getDir()+opts.appName+'.desktop'
         return file
 
-    enable: (opts, cb) ->
-        file = @getFile(opts)
-
-        data = [
-          '[Desktop Entry]',
-          'Type=Application',
-          'Vestion=1.0',
-          'Name='+opts.appName,
-          'Comment=' + opts.appName + ' startup script',
-          'Exec=' + opts.appPath,
-          'StartupNotify=false',
-          'Terminal=false'
-        ].join('\n')
-
-        mkdirp.sync(@getDir())
-        fs.writeFileSync(file, data)
-        cb()
-
-    disable: (opts, cb) ->
-        file = @getFile(opts)
-        if fs.existsSync(file)
-            fs.unlinkSync(file)
-        cb()
-
-    isEnabled: (opts, cb) ->
-        file = @getFile(opts)
-        cb(fs.existsSync(file))
+    enable: (opts) ->
+        new Promise (resolve, reject) =>
+            file = @getFile(opts)
+
+            data = [
+                '[Desktop Entry]',
+                'Type=Application',
+                'Vestion=1.0',
+                'Name='+opts.appName,
+                'Comment=' + opts.appName + ' startup script',
+                'Exec=' + opts.appPath,
+                'StartupNotify=false',
+                'Terminal=false'
+            ].join('\n')
+
+            mkdirp.sync(@getDir())
+            fs.writeFile file, data, (err) ->
+                return reject if err?
+                resolve()
+
+    disable: (opts) ->
+        new Promise (resolve, reject) =>
+            file = @getFile(opts)
+
+            fs.stat file, (err) ->
+                return reject(err) if err?
+                fs.unlink file, (err) ->
+                    return reject if err?
+                    resolve()
+
+    isEnabled: (opts) ->
+        new Promise (resolve, reject) =>
+            file = @getFile(opts)
+
+            fs.stat file, (err, stat) ->
+                resolve(stat?)
```

Comment:
I guess, considering we don't currently handle these types of errors well and of the amount of work involved; we can let it go in this pr. Maybe add a `# TODO: Error handling`. I'll add a separate issue to get this fixed. Would be good to be able to return consistent error messages regardless of the platform (when we can).

Thanks for the detailed response!